### PR TITLE
8272235: G1: update outdated code root fixup

### PR DESCRIPTION
**Pull request by albertnetymk**
Simple change of updating outdated comments and certain renaming.

Test: hotspot_gc

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8272235](https://bugs.openjdk.java.net/browse/JDK-8272235): G1: update outdated code root fixup

### Reviewers
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**) ⚠️ Review applies to bcd8b8e338708979c1515d10b90531a3a57c443f
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5073/head:pull/5073` \
`$ git checkout pull/5073`

Update a local copy of the PR: \
`$ git checkout pull/5073` \
`$ git pull https://git.openjdk.java.net/jdk pull/5073/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5073`

View PR using the GUI difftool: \
`$ git pr show -t 5073`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5073.diff">https://git.openjdk.java.net/jdk/pull/5073.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/g1/g1CollectedHeap.cpp [hunk 1]
```diff
@@ -3718,14 +3718,16 @@ void G1CollectedHeap::evacuate_initial_collection_set(G1ParScanThreadStateSet* p
                                       num_workers,
                                       has_optional_evacuation_work);
     task_time = run_task_timed(&g1_par_task);
-    // Closing the inner scope will execute the destructor for the G1RootProcessor object.
-    // To extract its code root fixup time we measure total time of this scope and
-    // subtract from the time the WorkGang task took.
+    // Closing the inner scope will execute the destructor for the
+    // G1RootProcessor object. By subtracting the WorkGang task from the total
+    // time of this scope, we get the "NMethod List Cleanup" time. Such list is
+    // constructed during "STW two-phase nmethod root processing", see more in
```
Comment (tschatzl) [inline src/hotspot/share/gc/g1/g1CollectedHeap.cpp:3724]: Maybe s/such/this.

**Review by tschatzl — APPROVED** [review]
Lgtm. I do not know why the "additional builds" would fail, but there is no log either.

File: src/hotspot/share/gc/g1/g1CollectedHeap.cpp (hunk 1 above)
Comment (albertnetymk) [reply src/hotspot/share/gc/g1/g1CollectedHeap.cpp:3724]: Fixed.

**Review by kimbarrett — APPROVED** [review]

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**